packages/plugin-auto-tracking-browser/src/auto-tracking-plugin.ts

```diff
@@ -0,0 +1,191 @@
+/* eslint-disable no-restricted-globals */
+import { BrowserClient, BrowserConfig, EnrichmentPlugin, IngestionMetadata } from '@amplitude/analytics-types';
+import * as constants from './constants';
+import { getText } from './helpers';
+
+type BrowserEnrichmentPlugin = EnrichmentPlugin<BrowserClient, BrowserConfig>;
+type ActionType = 'click' | 'change';
+
+const DEFAULT_TAG_ALLOWLIST = ['a', 'button', 'input', 'select', 'textarea', 'label'];
+
+interface EventListener {
+  element: Element;
+  type: ActionType;
+  handler: () => void;
+}
+
+interface Options {
+  cssSelectorAllowlist?: string[];
+  tagAllowlist?: string[];
+  ingestionMetadata?: IngestionMetadata; // Should avoid overriding this if unplanned to do so, this is not available in the charts.
```

Comment:
Add the ability to distinguish the ingestion source, this can be configured by the wrapped source like bookmarklet. This is only for internal usage.